meteor/server/api/playout/cache.ts

```diff
@@ -0,0 +1,271 @@
+import { Mutable, ReadonlyDeep } from 'type-fest'
+import _ from 'underscore'
+import { PartInstance, DBPartInstance, PartInstances } from '../../../lib/collections/PartInstances'
+import { Part, DBPart, Parts } from '../../../lib/collections/Parts'
+import { PeripheralDevice, PeripheralDevices } from '../../../lib/collections/PeripheralDevices'
+import { PieceInstance, PieceInstances } from '../../../lib/collections/PieceInstances'
+import {
+	RundownPlaylist,
+	DBRundownPlaylist,
+	RundownPlaylistId,
+	RundownPlaylists,
+} from '../../../lib/collections/RundownPlaylists'
+import { Rundown, DBRundown, Rundowns } from '../../../lib/collections/Rundowns'
+import { Segment, DBSegment, Segments } from '../../../lib/collections/Segments'
+import { Studio, StudioId, Studios } from '../../../lib/collections/Studios'
+import { Timeline, TimelineComplete } from '../../../lib/collections/Timeline'
+import { clone, waitForPromise } from '../../../lib/lib'
+import { ActivationCache, getActivationCache } from '../../cache/ActivationCache'
+import { DbCacheReadCollection, DbCacheWriteCollection } from '../../cache/CacheCollection'
+import { DbCacheReadObject, DbCacheWriteObject } from '../../cache/CacheObject'
+import { CacheBase, CacheForRundownPlaylist } from '../../cache/DatabaseCaches'
+import { profiler } from '../profiler'
+import { removeRundownPlaylistFromDb } from '../rundownPlaylist'
+import { CacheForStudioBase } from '../studio/cache'
+import { getRundownsSegmentsAndPartsFromCache } from './lib'
+
+export abstract class CacheForPlayoutPreInit extends CacheBase<CacheForPlayout> {
```

Comment:
doc: Could you write some documentation on what the intended use for the various caches are?

Reply:
done